indexer/services/ender/src/scripts/dydx_deleveraging_handler.sql

```diff
@@ -0,0 +1,134 @@
+    - perpetual_market: The perpetual market for the deleveraging in perpetual-market-model format (https://github.com/dydxprotocol/indexer/blob/cc70982/packages/postgres/src/models/perpetual-market-model.ts).
+    - liquidated_perpetual_position: The updated liquidated perpetual position in perpetual-position-model format (https://github.com/dydxprotocol/indexer/blob/cc70982/packages/postgres/src/models/perpetual-position-model.ts).
+    - offsetting_perpetual_position: The updated offsetting perpetual position in perpetual-position-model format (https://github.com/dydxprotocol/indexer/blob/cc70982/packages/postgres/src/models/perpetual-position-model.ts).
+*/
+CREATE OR REPLACE FUNCTION dydx_deleveraging_handler(
+    block_height int, block_time timestamp, event_data jsonb, event_index int, transaction_index int,
+    transaction_hash text) RETURNS jsonb AS $$
+DECLARE
+    QUOTE_CURRENCY_ATOMIC_RESOLUTION constant numeric = -6;
+    FEE constant numeric = 0;
+    perpetual_id bigint;
+    clob_pair_id bigint;
+    liquidated_subaccount_uuid uuid;
+    offsetting_subaccount_uuid uuid;
+    perpetual_market_record perpetual_markets%ROWTYPE;
+    liquidated_fill_record fills%ROWTYPE;
+    offsetting_fill_record fills%ROWTYPE;
+    liquidated_perpetual_position_record perpetual_positions%ROWTYPE;
+    offsetting_perpetual_position_record perpetual_positions%ROWTYPE;
+    liquidated_side text;
+    offsetting_side text;
+    size numeric;
+    price numeric;
+    event_id bytea;
+BEGIN
+    perpetual_id = (event_data->'perpetualId')::bigint;
+    BEGIN
+        SELECT * INTO STRICT perpetual_market_record FROM perpetual_markets WHERE "id" = perpetual_id;
+    EXCEPTION
+        WHEN NO_DATA_FOUND THEN
+            RAISE EXCEPTION 'Unable to find perpetual market with perpetualId %', perpetual_id;
+        WHEN TOO_MANY_ROWS THEN
+            /** This should never happen and if it ever were to would indicate that the table has malformed data. */
+            RAISE EXCEPTION 'Found multiple perpetual markets with perpetualId %', perpetual_id;
+    END;
+    /**
+      Calculate sizes, prices, and fill amounts.
+
+      TODO(IND-238): Extract out calculation of quantums and subticks to their own SQL functions.
+    */
+    size = dydx_trim_scale(dydx_from_jsonlib_long(event_data->'fillAmount') *
+                                 power(10, perpetual_market_record."atomicResolution")::numeric);
+    price = dydx_trim_scale(dydx_from_jsonlib_long(event_data->'price') *
+                                  power(10, QUOTE_CURRENCY_ATOMIC_RESOLUTION)::numeric);
+
+    liquidated_subaccount_uuid = dydx_uuid_from_subaccount_id(event_data->'liquidated');
+    offsetting_subaccount_uuid = dydx_uuid_from_subaccount_id(event_data->'offsetting');
+    offsetting_side = dydx_from_protocol_is_buy((event_data->'isBuy')::bool);
```

Comment:
Let us inline this function since I don't think `dydx_from_protocol_is_buy` adds a lot of value.
```suggestion
    offsetting_side = CASE WHEN (event_data->'isBuy')::bool THEN 'BUY' ELSE 'SELL' END;
```

Reply:
done